gnovm/pkg/gnolang/realm.go

```diff
@@ -213,8 +213,12 @@ func (rlm *Realm) DidUpdate(po, xo, co Object) {
 		if xo.GetRefCount() == 0 {
 			if xo.GetIsReal() {
 				rlm.MarkNewDeleted(xo)
+				// there's a chance xo is dirty and new deleted
+				// in one transaction, clear the dirty one once
+				// it's already marked new deleted.
+				rlm.clearDirtyNewDeleted()
```

Comment:
I think we want to avoid any non-O(1) operation in DidUpdate. Can we avoid this?

Reply:
moved this logic into processNewDeleted. While it will still be an O(n) operation due to traversing the slice, do you think this is acceptable?